src/main/drivers/system_stm32h7xx.c

```diff
@@ -183,18 +181,16 @@ void systemCheckResetReason(void)
         persistentObjectWrite(PERSISTENT_OBJECT_RESET_REASON, RESET_NONE);
         return;
 
-    case RESET_NONE:
-        if (!(RCC->RSR & RCC_RSR_SFTRSTF)) {
-            // Direct hard reset case
-            return;
-        }
-        // Soft reset; boot loader may have been active with BOOT pin pulled high.
-        FALLTHROUGH;
-
     case RESET_BOOTLOADER_POST:
         // Boot loader activity magically prevents SysTick from interrupting.
         // Issue a soft reset to prevent the condition.
         forcedSystemResetWithoutDisablingCaches(); // observed that disabling dcache after cold boot with BOOT pin high causes segfault.
+
+    case RESET_MSC_REQUEST:
```

Comment:
This is interesting - do you know why the problem was surfaced by the changes in #10247?

Reply:
Previously the configurator refused to invoke MSC as it said the hardware didn’t support it, so it never got this far.

Reply:
So this does not fix the reboot (not into MSC) issue listed in #10351?

Reply:
This fixes #10351 
RESET_MSC_REQUEST was not considered in this code.

Reply:
Interesting, this seems to be fixing both cases listed in #10351:
1. the flight controller entering DFU mode when `msc` is requested
2. the flight controller hanging on `exit`

Do you have any idea as to how this change relates to the changes in #10247? so that it fixes the hanging issue that did not exist prior to #10247??

Reply:
@mikeller This change is not applicable to the H7ZERO on which the USB MSC support was first added as `systemCheckResetReason()` is not called. On the IFLIGHT_H743_AIO the power on reset worked, but any soft reset failed as the RESET_NONE case would always fall through to call `forcedSystemResetWithoutDisablingCaches()` and thus get stuck in a reboot loop.

```
void SystemInit (void)
{
    memProtReset();

    initialiseMemorySections();

#if !defined(USE_EXST)
    // only stand-alone and bootloader firmware needs to do this.
    // if it's done in the EXST firmware as well as the BOOTLOADER firmware you get a reset loop.
    systemCheckResetReason();
#endif

```

Reply:
Thanks @SteveCEvans - always good to know _why_ a fix works. :sweat_smile: